### PR TITLE
Push notifications for reactions

### PR DESCRIPTION
**Pull request by basine**
**What's in this PR**

We are now creating push notifications for reactions for messages the selfUser sent.
We are also cancelling push notifications for "unlikes" from the same user.

Note: Creating pushes from message updateEvents needs refactoring. We should create notifications from messages instead of events to avoid parsing the event payload twice or more times.

### CODE REVIEW

File: Source/Notifications/Push notifications/Notification Types/ZMLocalNotificationForMessageEvent.swift [hunk 1]
```diff
@@ -216,3 +214,61 @@ public class ZMLocalNotificationForKnockMessage : ZMLocalNotificationForPostInCo
...
+        {
+            cancelNotifications()
+            shouldBeDiscarded = true
+            return nil
+        }
+        return nil
+    }
+    
+    override func configureAlertBody() -> String {
+        // TODO: use correct string
```
Comment (marcoconti83) [inline Source/Notifications/Push notifications/Notification Types/ZMLocalNotificationForMessageEvent.swift]: Todo?

**Comment by marcoconti83** [comment]
Looks good!

**Comment by marcoconti83** [comment]
About the comment ("we should refactor..."), please create a ticket to track it.

File: Source/Notifications/Push notifications/Notification Types/ZMLocalNotificationForMessageEvent.swift (hunk 1 above)
Comment (basine) [inline Source/Notifications/Push notifications/Notification Types/ZMLocalNotificationForMessageEvent.swift]: Right, that's Florian's comment. Will check if there is anything to do

**Pull request merged**